test/test_nn.py

```diff
@@ -3483,6 +3483,23 @@ def test_multihead_attn_all_arguments3():
             test_multihead_attn_all_arguments2()  # Test MultiheadAttention with all the argument.
         test_multihead_attn_all_arguments3()  # Test MultiheadAttention with all the argument.
 
+    def test_multihead_attn_3d_attn_mask(self):
+        embed_dim = 8
+        num_heads = 4
+        batch_size = 5
+        src_len = 3
+        tgt_len = 2
+
+        query = torch.rand(tgt_len, batch_size, embed_dim)
+        key = torch.rand(src_len, batch_size, embed_dim)
+        value = key
+        mta_model = torch.nn.MultiheadAttention(embed_dim, num_heads)
+        attn_mask_2d = torch.randint(0, 2, (tgt_len, src_len)).float()
+        attn_mask_2d = attn_mask_2d.masked_fill(attn_mask_2d == 0, float('-inf')).masked_fill(attn_mask_2d == 1, float(0.0)) 
+        attn_mask_3d = torch.stack([attn_mask_2d] * (batch_size * num_heads), dim=0)
```

Comment:
Instead of repeating the same 2d mask many times over I think it makes more sense to construct a list of different 2d masks and a 3d mask that covers all of them and then step through them (index into the results) one by one to make sure the 3d mask can cover the combined irregularity instead of passing a 3d mask that's just an expanded version of the 2d mask and making sure one entry is the same result.